src/components/EventDatesComponent/index.js

```diff
@@ -0,0 +1,215 @@
+    svg.classList.add("icon-1x1-small");
+    svg.setAttribute("aria-hidden", "true");
+    svg.setAttribute("role", "img");
+    svg.setAttribute("viewBox", "0 0 24 24");
+    svg.setAttribute("fill", "currentColor");
+    svg.innerHTML = `
+          <title>Time icon</title>
+          <path d="M12 22C6.477 22 2 17.523 2 12C2 6.477 6.477 2 12 2C17.523 2 22 6.477 22 12C22 17.523 17.523 22 12 22ZM12 20C14.1217 20 16.1566 19.1571 17.6569 17.6569C19.1571 16.1566 20 14.1217 20 12C20 9.87827 19.1571 7.84344 17.6569 6.34315C16.1566 4.84285 14.1217 4 12 4C9.87827 4 7.84344 4.84285 6.34315 6.34315C4.84285 7.84344 4 9.87827 4 12C4 14.1217 4.84285 16.1566 6.34315 17.6569C7.84344 19.1571 9.87827 20 12 20ZM13 12H17V14H11V7H13V12Z"/>
+          `;
+    return svg;
+  }
+
+  _createNextButton() {
+    const link = this._createElement("a", "event-date_next-button");
+    link.href = "#";
+    const svg = document.createElementNS("http://www.w3.org/2000/svg", "svg");
+    svg.classList.add("icon-1x1-small");
+    svg.setAttribute("width", "24");
+    svg.setAttribute("height", "24");
+    svg.setAttribute("viewBox", "0 0 24 24");
+    svg.setAttribute("fill", "none");
+    svg.innerHTML = `
+          <g clip-path="url(#clip0_396_1289)">
+            <path d="M13.1722 12L8.22217 7.04999L9.63617 5.63599L16.0002 12L9.63617 18.364L8.22217 16.95L13.1722 12Z" fill="currentColor"></path>
+          </g>
+          <defs>
+            <clipPath id="clip0_396_1289">
+              <rect width="24" height="24" fill="white"></rect>
+            </clipPath>
+          </defs>
+          `;
+    link.appendChild(svg);
+    return link;
+  }
+
+  _addSeeMoreAndNextButtonIfNeeded(component) {
+    component
+      .querySelectorAll(".event-date-cards_wrapper")
+      .forEach((wrapper) => {
+        const container = wrapper.querySelector(".event-date-cards_container");
+        if (container && container.scrollWidth > container.clientWidth) {
+          const nextButton = this._createNextButton(container);
+          nextButton.onclick = (e) => {
+            e.preventDefault();
+            container.scrollBy(260, 0);
+          };
+          wrapper.appendChild(nextButton);
+          wrapper.parentElement.appendChild(
+            this._createSeeMore((e) => {
+              e.preventDefault();
+              if (container.classList.contains("is-expanded")) {
+                container.classList.remove("is-expanded");
+                nextButton.style.display = "flex";
+                e.target.childNodes[0].nodeValue = "see more";
+                e.target.childNodes[1].style.transform = "rotate(0deg)";
+              } else {
+                container.classList.add("is-expanded");
+                nextButton.style.display = "none";
+                e.target.childNodes[0].nodeValue = "see less";
+                e.target.childNodes[1].style.transform = "rotate(180deg)";
+              }
+            })
```

Comment:
Let's move the handler function to a variable. It would easier to read.
```suggestion
const onClickSeeMore = (e) => {
  e.preventDefault();
  if (container.classList.contains("is-expanded")) {
    container.classList.remove("is-expanded");
    nextButton.style.display = "flex";
    e.target.childNodes[0].nodeValue = "see more";
    e.target.childNodes[1].style.transform = "rotate(0deg)";
  } else {
    container.classList.add("is-expanded");
    nextButton.style.display = "none";
    e.target.childNodes[0].nodeValue = "see less";
    e.target.childNodes[1].style.transform = "rotate(180deg)";
  }
};

wrapper.parentElement.appendChild(this._createSeeMore(onClickSeeMore));
```